### PR TITLE
feat: add util methods for rows and cells to GridElement (#5508)

### LINKED ISSUE #5460: Testbench GridTRElement class should provide a method to get all cells of row
### Describe your motivation

We have use cases where we need to test if for example a record has been sucessfully entered into the database. The data from the database is displayed in a grid. We would like to fetch all cells of a row with one single call. 

### Describe the solution you'd like

It would be nice if `com.vaadin.flow.component.grid.testbench.GridTRElement` had a method to fetch all cells of a row, something like:
`public List<GridTHTDElement> getCells(GridColumnElement... column){...}` 

The argument(s) could be an array or some sort of collection of type `com.vaadin.flow.component.grid.testbench.GridColumnElement`. Also the return value could be an array or some sort of collection of type `com.vaadin.flow.component.grid.testbench.GridTHTDElement`. 

### Describe alternatives you've considered

Calling the existing `com.vaadin.flow.component.grid.testbench.GridTRElement#getCell` method for each column.

### Additional context

should be available for vaadin 14

### PR DESCRIPTION
**Pull request by rodolforfq**
## Description

Based on #5508: add util methods to Testbench's GridElement class to allow for easier testing in cases where full rows and cells details/contents are needed.

Fixes #5460 and #5461 

## Type of change

- [ ] Bugfix
- [X] Feature

## Checklist

- [X] I have read the contribution guide: https://vaadin.com/docs/latest/contributing/overview
- [X] I have added a description following the guideline.
- [X] The issue is created in the corresponding repository and I have referenced it.
- [X] I have added tests to ensure my change is effective and works as intended.
- [X] New and existing tests are passing locally with my change.
- [X] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [X] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

**Comment by ZheSun88** [comment]
let us update this PR target to 23.4 after #5656 got merged.

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridTestbenchUtilIT.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    @Before
+    public void init() {
+        open();
+        grid = $(GridElement.class).first();
+    }
+
+    @Test
+    public void cellsRowIndexUtilMethod_returnsExpectedCells() {
+        IntStream.rangeClosed(grid.getFirstVisibleRowIndex(),
+                grid.getLastVisibleRowIndex()).forEach(row -> {
```
Comment (tomivirkki) [inline vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridTestbenchUtilIT.java:49]: This file seems to be from an earlier revision before [these comments](https://github.com/vaadin/flow-components/pull/5508#discussion_r1352095602) were addressed.

**Comment by tomivirkki** [comment]
Some integration tests are failing:

Screenshot 2023-11-08 at 17 38 21

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridTestbenchUtilIT.java (hunk 1 above)
Comment (rodolforfq) [reply vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridTestbenchUtilIT.java:49]: Will take a look at this

**Review by tomivirkki — APPROVED** [review]

**Comment by rodolforfq** [comment]
All should be ok now @tomivirkki

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**